libyang/context.py

```diff
@@ -468,3 +511,41 @@ def __iter__(self) -> Iterator[Module]:
         while mod:
             yield Module(self, mod)
             mod = lib.ly_ctx_get_module_iter(self.cdata, idx)
+
+    def free_module_data(self, cdata) -> None:
+        self._cdata_modules.remove(cdata)
+
+    def get_module_data(
+        self,
+        mod_name: Optional[str],
+        mod_rev: Optional[str],
+        submod_name: Optional[str],
+        submod_rev: Optional[str],
+    ) -> Tuple[str, Optional[str]]:
+        if self._module_data_clb is None:
+            return None
+        fmt_str, module_data = self._module_data_clb(
+            mod_name, mod_rev, submod_name, submod_rev
+        )
+        if module_data is None:
+            return fmt_str, None
+        module_data_c = str2c(module_data)
+        self._cdata_modules.append(module_data_c)
```

Comment:
Why do you need to store this?

Reply:
The data provided are python string. That one needs to held as libyang is using those data directly. It doesn't create internal copy. Therefore the data must stay in memory until libyang will ask to free them. If it would not be held by context itself, it will be up to user to hold them and release them when not needed anymore